homeassistant/helpers/intent.py

```diff
@@ -68,18 +69,26 @@ def async_register(hass: HomeAssistant, handler: IntentHandler) -> None:
 
     intents[handler.intent_type] = handler
 
+    if handler.intent_type not in LLM_SKIP_AUTO_ADD_TOOL:
+        llm_tool = IntentTool(handler.intent_type, handler.effective_slot_schema)
+        try:
+            llm.async_register_tool(hass, llm_tool)
```

Comment:
The intent helper shouldn't be aware of LLM tools at all. When LLM tools are fetched, it should generate the spec for the intents that are registered at that moment instead.

Reply:
Done